butterfree/extract/readers/reader.py

```diff
@@ -139,3 +144,45 @@ def _apply_transformations(self, df: DataFrame) -> DataFrame:
             self.transformations,
             df,
         )
+
+    def build_metadata(
+        self,
+    ) -> Union[FileReaderMetadata, KafkaReaderMetadata, TableReaderMetadata]:
+        """Get the reader's metadata as a Pydantic model.
+
+        This method creates a standardized representation of reader metadata
+        that can be used for documentation, validation, and serialization purposes.
+        Each reader type (File, Kafka, Table) has its own specific metadata
+        while sharing common base attributes.
+
+        Returns:
+            A BaseReaderMetadata model containing the reader's metadata
+        """
+        reader_type_map = {
+            "FileReader": FileReaderMetadata,
+            "KafkaReader": KafkaReaderMetadata,
+            "TableReader": TableReaderMetadata,
+        }
```

Comment:
minor: it feels a bit backwards that the base/abstract class needs to know the derived classes for generating metadata. Any thoughts?